src/pocketmine/Player.php

```diff
@@ -4372,7 +4372,9 @@ public function attack(EntityDamageEvent $source) : void{
 			and $source->getCause() !== EntityDamageEvent::CAUSE_VOID
 		){
 			$source->setCancelled();
-		}elseif($this->allowFlight and $source->getCause() === EntityDamageEvent::CAUSE_FALL){
+		}elseif(($this->isCreative() || $this->isSpectaror())
```

Comment:
isSpectaror() to $this->isSpectator() plz

Reply:
Fly still needs another change from the .20-.40 to work in sync with mc

Reply:
fly and falldamage is broken since the remove of the adventure settings

Reply:
https://github.com/Benedikt05/BetterAltay/commit/d016f393feea29938e6c20efcf1d2c167f37d595

Reply:
it work if you remove $this->allowFlight in line 4376

Reply:
but the rest is perfekt jose thx for the fast update

Reply:
The isSpectator function is not written correctly